### PR TITLE
Interpret XML strictly

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's another behavior break when you get a chance

File: lib/common/xml_io.c [hunk 1]
```diff
@@ -24,46 +24,6 @@
 #include <crm/common/xml_io.h>
 #include "crmcommon_private.h"
 
-/* @COMPAT XML_PARSE_RECOVER allows some XML errors to be silently worked around
```
Comment (nrwahl2) [inline lib/common/xml_io.c:27]: I thought I remembered that *for some reason* we had decided to wait on this beyond 3.0.0... But I have no idea why that might have been, and as of https://github.com/ClusterLabs/pacemaker/pull/3361#discussion_r1490096446 we were still talking about dropping it. So I'm happy to assume I'm misremembering.

File: cts/cts-cli.in [hunk 2]
```diff
@@ -1276,7 +1276,7 @@ class CibadminRegressionTest(RegressionTest):
 
         # Add some stuff to the empty CIB so we know that erasing it did something.
         basic_tests_setup = [
-            """cibadmin -C -o nodes --xml-text '<node id="node1" uname="node1">'""",
+            """cibadmin -C -o nodes --xml-text '<node id="node1" uname="node1"/>'""",
```
Comment (nrwahl2) [inline cts/cts-cli.in:1279]: I'm surprised the outputs don't change... I would've guessed an error on first parse and then success on second parse. Oh well.

**Review by nrwahl2 — APPROVED** [review]

File: lib/common/xml_io.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/xml_io.c:27]: We hadn't expected that something like `cibadmin -C -o nodes --xml-text '<node id="node1" uname="node1">'` (missing close at end of tag) would stop working. But pcs and crm shell do not appear to do anything like that, so we're better off moving forward.

File: cts/cts-cli.in (hunk 2 above)
Comment (kgaillot) [reply cts/cts-cli.in:1279]: There are two aspects to the strict interpretation: accepting partial XML and accepting recoverably invalid XML. The retry was for the second category, but this one fell into the first. If we stopped accepting partial XML (only), it *would* be detected and recovered as invalid XML, but accepting the partial XML meant the recovery never kicked in.

**Pull request merged**